lib/document-to-html.js

```diff
@@ -100,21 +100,55 @@ function DocumentConversion(options, comments) {
             return [];
         }
     }
+
+    function enrichListItemAttrs(num, li) {
+        if(li==null && li.tag==null || li.tag.tagName!=='li') return;
+
+        Object.assign(li.tag.attributes, {'data-numbering': num});
+    }
+
+    function traverseTreeForLI(el, num, shouldApply){
+        if(el==null) return;
+        
+        if(Array.isArray(el)) {
+            el.forEach(subEl=> traverseTreeForLI(subEl, num, shouldApply));
+        }
+        
+        if(!Array.isArray(el) && Array.isArray(el.children)) {
+            var apply = el.tag!=null && el.tag.tagName==='ol';
+            traverseTreeForLI(el.children, num, apply);
+        }
+
+        if(el!=null && el.tag!=null && el.tag.tagName==='li' && shouldApply) {
+            enrichListItemAttrs(num, el);
+        }
+    }
+
+    function enrichNumberingAttrs(out, element) {
+        if(element.numbering==null || !Array.isArray(out)) return;
+        var num = JSON.stringify(element.numbering.attrs).replace(/"/g,"'");
+
+        traverseTreeForLI(out, num, false);
+    }
     
     function convertParagraph(element, messages, options) {
-        return  htmlPathForParagraph(element, messages).wrap(function() {
-            var content = convertElements(element.children, messages, options);
-            if (ignoreEmptyParagraphs) {
-                return content;
-            } else {
-                return [Html.forceWrite].concat(content);
-            }
-        });
+        var out = htmlPathForParagraph(element, messages)
+            .wrap(function() {
+                var content = convertElements(element.children, messages, options);
+                if (ignoreEmptyParagraphs) {
+                    return content;
+                } else {
+                    return [Html.forceWrite].concat(content);
+                }
+            });
+
+        enrichNumberingAttrs(out, element);
```

Comment:
I tried to work out how to enrich the elements during creation but wow the style rule system is a bit of a brute.
Seems like the attributes would have to be passed to `findStyle` below in order to get the attrs added during element creation but the effort needed seems excessive!

This approach seems reasonable given the constraints of the codebase

Reply:
That's not the only problem, even if we add them in findStyle, we'd need to copy those instances because they are reused for all elements which match that style.